HeterogeneousCore/CUDACore/interface/CUDAScopedContext.h

```diff
@@ -27,22 +29,20 @@ namespace impl {
     // mutable access is needed even if the CUDAScopedContext itself
     // would be const. Therefore it is ok to return a non-const
     // pointer from a const method here.
-    cudaStream_t stream() const { return stream_->id(); }
-    const std::shared_ptr<cuda::stream_t<>>& streamPtr() const { return stream_; }
+    cudaStream_t stream() const { return stream_.get(); }
+    const cudautils::SharedStreamPtr& streamPtr() const { return stream_; }
 
   protected:
     explicit CUDAScopedContextBase(edm::StreamID streamID);
 
     explicit CUDAScopedContextBase(const CUDAProductBase& data);
 
-    explicit CUDAScopedContextBase(int device, std::shared_ptr<cuda::stream_t<>> stream);
-
-    std::shared_ptr<cuda::stream_t<>>& streamPtr() { return stream_; }
```

Comment:
this is not needed any more ?

Reply:
Right, not needed anymore as the `const` variant above is enough.